### PR TITLE
enclose statement in %isdarwin conditional

### PR DESCRIPTION
**Pull request by gartung**
@davidlt @smuzaffar @Dr15Jones

This edit cause a segmentation fault in the linux version of this library when this is user

### CODE REVIEW

**Comment by davidlt** [comment]
Could you explain in more details? How Darwin specific (or maybe it's not?) file is affecting Linux?

**Comment by gartung** [comment]
Run 'scram b -v dxr' with CMSSW_7_6_0_pre6 on slc6 and you will see that a segfault happens while loading libtrilite.so. Removing this line from the spec file and rebuilding is the only change that removed the segault.

**Comment by gartung** [comment]
[gartung@cmssrv208 CMSSW_7_6_0_pre6]$ scram setup ~/cmsbuild/slc6_amd64_gcc493/external/py2-dxr-toolfile/1.0-kpegke2/etc/scram.d/py2-dxr.xml 

Finding a value for PY2_DXR_BASE:

```
Checks [OK] for /storage/local/data1/gartung/cmsbuild/slc6_amd64_gcc493/external/py2-dxr/1.0-kpegke2
```

Finding a value for LIBDIR:

```
Checks [OK] for /storage/local/data1/gartung/cmsbuild/slc6_amd64_gcc493/external/py2-dxr/1.0-kpegke2/lib
```

---

Runtime path settings for PYTHONPATH:

```
Checks [OK] for /storage/local/data1/gartung/cmsbuild/slc6_amd64_gcc493/external/py2-dxr/1.0-kpegke2/lib/python2.7/site-packages
```

Runtime path settings for PATH:

```
Checks [OK] for /storage/local/data1/gartung/cmsbuild/slc6_amd64_gcc493/external/py2-dxr/1.0-kpegke2/bin
```

Setting up py2-dxr version 1.0-kpegke2:  
Updating symlinks under external/slc6_amd64_gcc493
[gartung@cmssrv208 CMSSW_7_6_0_pre6]$ scram b -v dxr
Reading cached build data

> > Local Products Rules ..... started
> > Local Products Rules ..... done
> > ***_WARNING: Invalid tool DataFormats/MuonData. Please fix src/HLTrigger/HLTanalyzers/BuildFile.xml file.
> > Local Products Rules ..... started
> > Local Products Rules ..... done
> > ***_WARNING: Invalid tool DataFormats/MuonData. Please fix src/HLTrigger/HLTanalyzers/BuildFile.xml file.
> > /bin/sh: line 1:  6428 Segmentation fault      SCRAM_DXR_RUN=yes /storage/local/data1/gartung/cmsbuild/slc6_amd64_gcc493/external/py2-dxr/1.0-kpegke2/bin/dxr-build.py -v -f src/dxr.config
> > gmake: **\* [dxr] Error 139
> > gmake: **\* [There are compilation/build errors. Please see the detail log above.] Error 2

**Comment by gartung** [comment]
[gartung@cmssrv208 CMSSW_7_6_0_pre6]$ gdb -ex r --args python /storage/local/data1/gartung/cmsbuild/slc6_amd64_gcc493/external/py2-dxr/1.0-kpegke2/bin/dxr-build.py -v -f src/dxr.config
GNU gdb (GDB) 7.9.1
Copyright (C) 2015 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later http://gnu.org/licenses/gpl.html
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-unknown-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
http://www.gnu.org/software/gdb/bugs/.
Find the GDB manual and other documentation resources online at:
http://www.gnu.org/software/gdb/documentation/.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from python...done.
Starting program: /storage/local/data1/cms_admin/sw/slc6_amd64_gcc493/cms/cmssw/CMSSW_7_6_0_pre6/external/slc6_amd64_gcc493/bin/python /storage/local/data1/gartung/cmsbuild/slc6_amd64_gcc493/external/py2-dxr/1.0-kpegke2/bin/dxr-build.py -v -f src/dxr.config
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
warning: File "/storage/local/data1/cms_admin/sw/slc6_amd64_gcc493/external/gcc/4.9.3/lib64/libstdc++.so.6.0.20-gdb.py" auto-loading has been declined by your `auto-load safe-path' set to "$debugdir:$datadir/auto-load".
To enable execution of this file add
    add-auto-load-safe-path /storage/local/data1/cms_admin/sw/slc6_amd64_gcc493/external/gcc/4.9.3/lib64/libstdc++.so.6.0.20-gdb.py
line to your configuration file "/storage/local/data1/gartung/.gdbinit".
To completely disable this security protection add
    set auto-load safe-path /
line to your configuration file "/storage/local/data1/gartung/.gdbinit".
For more information about this security protection see the
"Auto-loading safe path" section in the GDB manual.  E.g., run from the shell:
    info "(gdb)Auto-loading safe path"

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff798ebf1 in load_trilite_extension ()
   from /storage/local/data1/gartung/CMSSW_7_6_0_pre6/external/slc6_amd64_gcc493/lib/libtrilite.so
(gdb) bt
#0  0x00007ffff798ebf1 in load_trilite_extension ()
   from /storage/local/data1/gartung/CMSSW_7_6_0_pre6/external/slc6_amd64_gcc493/lib/libtrilite.so
#1  0x00007ffff7a05900 in ffi_call_unix64 ()
    at /build/cmsbuild/auto-builds/CMSSW_7_6_0_pre5-slc6_amd64_gcc493/build/CMSSW_7_6_0_pre5-build/BUILD/slc6_amd64_gcc493/external/python/2.7.6-kpegke/python-2.7.6/Modules/_ctypes/libffi/src/x86/unix64.S:76
#2  0x00007ffff7a04a78 in ffi_call (cif=cif@entry=0x7fffffff35f0, fn=fn@entry=0x7ffff798ebe0 <load_trilite_extension>, 
    rvalue=rvalue@entry=0x7fffffff3550, avalue=avalue@entry=0x7fffffff3540)
    at /build/cmsbuild/auto-builds/CMSSW_7_6_0_pre5-slc6_amd64_gcc493/build/CMSSW_7_6_0_pre5-build/BUILD/slc6_amd64_gcc493/external/python/2.7.6-kpegke/python-2.7.6/Modules/_ctypes/libffi/src/x86/ffi64.c:522
#3  0x00007ffff79fc6f2 in _call_function_pointer (argcount=0, resmem=0x7fffffff3550, restype=<optimized out>, atypes=<optimized out>, 
    avalues=0x7fffffff3540, pProc=0x7ffff798ebe0 <load_trilite_extension>, flags=4353)
    at /build/cmsbuild/auto-builds/CMSSW_7_6_0_pre5-slc6_amd64_gcc493/build/CMSSW_7_6_0_pre5-build/BUILD/slc6_amd64_gcc493/external/python/2.7.6-kpegke/python-2.7.6/Modules/_ctypes/callproc.c:836
#4  _ctypes_callproc (pProc=pProc@entry=0x7ffff798ebe0 <load_trilite_extension>, argtuple=argtuple@entry=0x7ffff7d98050, flags=4353, 
    argtypes=argtypes@entry=0x0, restype=0xb659c0, checker=0x0)
    at /build/cmsbuild/auto-builds/CMSSW_7_6_0_pre5-slc6_amd64_gcc493/build/CMSSW_7_6_0_pre5-build/BUILD/slc6_amd64_gcc493/external/python/2.7.6-kpegke/python-2.7.6/Modules/_ctypes/callproc.c:1183
#5  0x00007ffff79f36f4 in PyCFuncPtr_call (self=<optimized out>, inargs=<optimized out>, kwds=<optimized out>)
    at /build/cmsbuild/auto-builds/CMSSW_7_6_0_pre5-slc6_amd64_gcc493/build/CMSSW_7_6_0_pre5-build/BUILD/slc6_amd64_gcc493/external/python/2.7.6-kpegke/python-2.7.6/Modules/_ctypes/_ctypes.c:3929
#6  0x00007ffff7e48133 in PyObject_Call (func=func@entry=0x9c5e20, arg=arg@entry=0x7ffff7d98050, kw=kw@entry=0x0) at Objects/abstract.c:2529
#7  0x00007ffff7efc40e in do_call (nk=<optimized out>, na=0, pp_stack=0x7fffffff3830, func=0x9c5e20) at Python/ceval.c:4239
#8  call_function (oparg=<optimized out>, pp_stack=0x7fffffff3830) at Python/ceval.c:4044
#9  PyEval_EvalFrameEx (f=f@entry=0xb581b0, throwflag=throwflag@entry=0) at Python/ceval.c:2666
#10 0x00007ffff7eff6a0 in PyEval_EvalCodeEx (co=co@entry=0xa3a430, globals=globals@entry=0xae2720, locals=locals@entry=0xae2720, 
    args=args@entry=0x0, argcount=argcount@entry=0, kws=kws@entry=0x0, kwcount=0, defs=0x0, defcount=0, closure=0x0) at Python/ceval.c:3253
#11 0x00007ffff7eff7c9 in PyEval_EvalCode (co=co@entry=0xa3a430, globals=globals@entry=0xae2720, locals=locals@entry=0xae2720)
    at Python/ceval.c:667
#12 0x00007ffff7f12ed9 in PyImport_ExecCodeModuleEx (name=name@entry=0x4f01f0 "dxr.utils", co=co@entry=0xa3a430, 
    pathname=pathname@entry=0x6bc7d0 "/storage/local/data1/gartung/cmsbuild/slc6_amd64_gcc493/external/py2-dxr/1.0-kpegke2/lib/python2.7/site-packages/dxr/utils.pyc") at Python/import.c:709
#13 0x00007ffff7f1322e in load_source_module (name=0x4f01f0 "dxr.utils", 
    pathname=0x6bc7d0 "/storage/local/data1/gartung/cmsbuild/slc6_amd64_gcc493/external/py2-dxr/1.0-kpegke2/lib/python2.7/site-packages/dxr/utils.pyc", fp=<optimized out>) at Python/import.c:1099
#14 0x00007ffff7f14019 in import_submodule (mod=mod@entry=0x643088, subname=subname@entry=0x4f01f4 "utils", 
    fullname=fullname@entry=0x4f01f0 "dxr.utils") at Python/import.c:2700
#15 0x00007ffff7f14d44 in load_next (p_buflen=<synthetic pointer>, buf=0x4f01f0 "dxr.utils", p_name=<synthetic pointer>, altmod=0x643088, 
    mod=0x643088) at Python/import.c:2515
#16 import_module_level (locals=<optimized out>, level=<optimized out>, fromlist=<optimized out>, globals=<optimized out>, 
    name=<optimized out>) at Python/import.c:2232
---Type <return> to continue, or q <return> to quit--- 
#17 PyImport_ImportModuleLevel (name=<optimized out>, globals=<optimized out>, locals=<optimized out>, fromlist=<optimized out>, 
    level=<optimized out>) at Python/import.c:2288
#18 0x00007ffff7ef67a8 in builtin___import__ (self=<optimized out>, args=<optimized out>, kwds=<optimized out>) at Python/bltinmodule.c:49
#19 0x00007ffff7e48133 in PyObject_Call (func=func@entry=0x7ffff7dc6050, arg=arg@entry=0x5e24c8, kw=<optimized out>)
    at Objects/abstract.c:2529
#20 0x00007ffff7ef82e7 in PyEval_CallObjectWithKeywords (func=func@entry=0x7ffff7dc6050, arg=arg@entry=0x5e24c8, kw=kw@entry=0x0)
    at Python/ceval.c:3890
#21 0x00007ffff7efa051 in PyEval_EvalFrameEx (f=f@entry=0x6680c0, throwflag=throwflag@entry=0) at Python/ceval.c:2333
#22 0x00007ffff7eff6a0 in PyEval_EvalCodeEx (co=co@entry=0x5e57b0, globals=globals@entry=0x679a80, locals=locals@entry=0x679a80, 
    args=args@entry=0x0, argcount=argcount@entry=0, kws=kws@entry=0x0, kwcount=0, defs=0x0, defcount=0, closure=0x0) at Python/ceval.c:3253
#23 0x00007ffff7eff7c9 in PyEval_EvalCode (co=co@entry=0x5e57b0, globals=globals@entry=0x679a80, locals=locals@entry=0x679a80)
    at Python/ceval.c:667
#24 0x00007ffff7f12ed9 in PyImport_ExecCodeModuleEx (name=name@entry=0x6200b0 "dxr.build", co=co@entry=0x5e57b0, 
    pathname=pathname@entry=0x61e470 "/storage/local/data1/gartung/cmsbuild/slc6_amd64_gcc493/external/py2-dxr/1.0-kpegke2/lib/python2.7/site-packages/dxr/build.pyc") at Python/import.c:709
#25 0x00007ffff7f1322e in load_source_module (name=0x6200b0 "dxr.build", 
    pathname=0x61e470 "/storage/local/data1/gartung/cmsbuild/slc6_amd64_gcc493/external/py2-dxr/1.0-kpegke2/lib/python2.7/site-packages/dxr/build.pyc", fp=<optimized out>) at Python/import.c:1099
#26 0x00007ffff7f14019 in import_submodule (mod=mod@entry=0x643088, subname=subname@entry=0x6200b4 "build", 
    fullname=fullname@entry=0x6200b0 "dxr.build") at Python/import.c:2700
#27 0x00007ffff7f14d44 in load_next (p_buflen=<synthetic pointer>, buf=0x6200b0 "dxr.build", p_name=<synthetic pointer>, altmod=0x643088, 
    mod=0x643088) at Python/import.c:2515
#28 import_module_level (locals=<optimized out>, level=<optimized out>, fromlist=<optimized out>, globals=<optimized out>, 
    name=<optimized out>) at Python/import.c:2232
#29 PyImport_ImportModuleLevel (name=<optimized out>, globals=<optimized out>, locals=<optimized out>, fromlist=<optimized out>, 
    level=<optimized out>) at Python/import.c:2288
#30 0x00007ffff7ef67a8 in builtin___import__ (self=<optimized out>, args=<optimized out>, kwds=<optimized out>) at Python/bltinmodule.c:49
#31 0x00007ffff7e48133 in PyObject_Call (func=func@entry=0x7ffff7dc6050, arg=arg@entry=0x532158, kw=<optimized out>)
    at Objects/abstract.c:2529
#32 0x00007ffff7ef82e7 in PyEval_CallObjectWithKeywords (func=func@entry=0x7ffff7dc6050, arg=arg@entry=0x532158, kw=kw@entry=0x0)
    at Python/ceval.c:3890
#33 0x00007ffff7efa051 in PyEval_EvalFrameEx (f=f@entry=0x5c9500, throwflag=throwflag@entry=0) at Python/ceval.c:2333
#34 0x00007ffff7eff6a0 in PyEval_EvalCodeEx (co=co@entry=0x634230, globals=globals@entry=0x480d20, locals=locals@entry=0x480d20, 
    args=args@entry=0x0, argcount=argcount@entry=0, kws=kws@entry=0x0, kwcount=0, defs=0x0, defcount=0, closure=0x0) at Python/ceval.c:3253
#35 0x00007ffff7eff7c9 in PyEval_EvalCode (co=co@entry=0x634230, globals=globals@entry=0x480d20, locals=locals@entry=0x480d20)
    at Python/ceval.c:667
#36 0x00007ffff7f234da in run_mod (arena=0x427a30, flags=0x7fffffff4390, locals=0x480d20, globals=0x480d20, filename=0x55f2c0 "\001", 
    mod=<optimized out>) at Python/pythonrun.c:1370
#37 PyRun_FileExFlags (fp=fp@entry=0x55f2c0, 
    filename=filename@entry=0x7fffffff4bce "/storage/local/data1/gartung/cmsbuild/slc6_amd64_gcc493/external/py2-dxr/1.0-kpegke2/bin/dxr-build---Type <return> to continue, or q <return> to quit---
.py", start=start@entry=257, globals=globals@entry=0x480d20, locals=locals@entry=0x480d20, closeit=closeit@entry=1, flags=0x7fffffff4390)
    at Python/pythonrun.c:1356
#38 0x00007ffff7f24a57 in PyRun_SimpleFileExFlags (fp=fp@entry=0x55f2c0, 
    filename=0x7fffffff4bce "/storage/local/data1/gartung/cmsbuild/slc6_amd64_gcc493/external/py2-dxr/1.0-kpegke2/bin/dxr-build.py", 
    closeit=1, flags=flags@entry=0x7fffffff4390) at Python/pythonrun.c:948
#39 0x00007ffff7f24f63 in PyRun_AnyFileExFlags (fp=fp@entry=0x55f2c0, filename=<optimized out>, closeit=<optimized out>, 
    flags=flags@entry=0x7fffffff4390) at Python/pythonrun.c:752
#40 0x00007ffff7f3aab5 in Py_Main (argc=<optimized out>, argv=<optimized out>) at Modules/main.c:640
#41 0x00000039d301ed5d in __libc_start_main () from /lib64/libc.so.6
#42 0x00000000004006b1 in _start ()

**Comment by gartung** [comment]
The version that works has an additional symbols sqlite3_auto_extension

[gartung@cmssrv208 py2-dxr]$ nm -C 1.0-cms6/lib/libtrilite.so | grep sql
00000000000747c8 B sqlite3_api
                 U sqlite3_auto_extension
000000000001abf0 T sqlite3_extension_init
[gartung@cmssrv208 py2-dxr]$ nm -C 1.0-kpegke2/lib/libtrilite.so | grep sql
00000000000747a8 B sqlite3_api
000000000001ab90 T sqlite3_extension_init

**Comment by gartung** [comment]
Ah... the working and non-working version are built against different sqlite versions...

[gartung@cmssrv208 py2-dxr]$ diff 1.0-kpegke2/spec  1.0-cms6/spec

27,31c27,31
< %define buildpkgreqs external/llvm/3.6-kpegke2
< %define builddirectpkgreqs external/llvm/3.6-kpegke2
< %define allpkgreqs   external/gcc/4.9.3 external/bz2lib/1.0.5 external/db4/4.4.20 external/expat/2.0.1 external/gdbm/1.10 external/ncurses/5.9 external/openssl/0.9.8e__1.0.1 external/readline/6.2 external/sqlite/3.8.11.1 external/zlib/1.2.8 external/python/2.7.6-kpegke external/py2-setuptools/2.1-kpegke external/llvm/3.6-kpegke2 external/py2-futures/2.2.0-kpegke external/py2-jinja/2.7.2-kpegke external/py2-markupsafe/0.23-kpegke external/py2-ordereddict/1.1-kpegke external/py2-parsimonious/0.6.1-kpegke
< Requires: external+gcc+4.9.3 external+bz2lib+1.0.5 external+db4+4.4.20 external+expat+2.0.1 external+gdbm+1.10 external+ncurses+5.9 external+openssl+0.9.8e__1.0.1 external+readline+6.2 external+sqlite+3.8.11.1 external+zlib+1.2.8 external+python+2.7.6-kpegke external+py2-setuptools+2.1-kpegke external+py2-futures+2.2.0-kpegke external+py2-jinja+2.7.2-kpegke external+py2-markupsafe+0.23-kpegke external+py2-ordereddict+1.1-kpegke external+py2-parsimonious+0.6.1-kpegke

## < BuildRequires: external+llvm+3.6-kpegke2

> %define buildpkgreqs external/llvm/3.6
> %define builddirectpkgreqs external/llvm/3.6
> %define allpkgreqs   external/gcc/4.9.3 external/bz2lib/1.0.5 external/db4/4.4.20 external/expat/2.0.1 external/gdbm/1.10 external/ncurses/5.9 external/openssl/0.9.8e__1.0.1 external/readline/6.2 external/sqlite/3.7.17 external/zlib/1.2.8 external/python/2.7.6 external/py2-setuptools/2.1 external/llvm/3.6 external/py2-futures/2.2.0 external/py2-jinja/2.7.2 external/py2-markupsafe/0.23 external/py2-ordereddict/1.1 external/py2-parsimonious/0.6.1
> Requires: external+gcc+4.9.3 external+bz2lib+1.0.5 external+db4+4.4.20 external+expat+2.0.1 external+gdbm+1.10 external+ncurses+5.9 external+openssl+0.9.8e__1.0.1 external+readline+6.2 external+sqlite+3.7.17 external+zlib+1.2.8 external+python+2.7.6 external+py2-setuptools+2.1 external+py2-futures+2.2.0 external+py2-jinja+2.7.2 external+py2-markupsafe+0.23 external+py2-ordereddict+1.1 external+py2-parsimonious+0.6.1
> BuildRequires: external+llvm+3.6
> 36,37c36,37
> < Provides: external+py2-dxr+1.0-kpegke2
> 
> ## < Obsoletes: external+py2-dxr+1.0-kpegke2
> 
> Provides: external+py2-dxr+1.0-cms6
> Obsoletes: external+py2-dxr+1.0-cms6

**Comment by davidlt** [comment]
This needs to be debugged, but that's wrong:

```
=> 0x00007fffefcfcbf1 <+17>:    mov    0x600(%rax),%rax
..
rax            0x0      0
```

```
==6837== Process terminating with default action of signal 11 (SIGSEGV)
==6837==  Access not within mapped region at address 0x600
==6837==    at 0xEA1DBF1: load_trilite_extension (in /afs/cern.ch/cms/sw/ReleaseCandidates/volC/slc6_amd64_gcc493/external/py2-dxr/1.0-kpegke2/lib/libtrilite.so)
==6837==    by 0x41C48FF: ffi_call_unix64 (unix64.S:76)
```

```
==6837== Invalid read of size 8
==6837==    at 0xEA1DBF1: load_trilite_extension (in /afs/cern.ch/cms/sw/ReleaseCandidates/volC/slc6_amd64_gcc493/external/py2-dxr/1.0-kpegke2/lib/libtrilite.so)
```

I hate this package, because it's a hacky mess. I think, I had to patch it again for Clang 3.7.0.

Probably this is the place where we need to start looking.

`dxr/utils.py`

```
  1 import ctypes
  2
  3 # Load the trilite plugin.
  4 #
  5 # If you ``import sqlite3`` before doing this, it's likely that the system
  6 # version of sqlite will be loaded, and then trilite, if built against a
  7 # different version, will fail to load. If you're having trouble getting
  8 # trilite to load, make sure you're not importing sqlite3 beforehand. Afterward
  9 # is fine.
 10 ctypes.CDLL('libtrilite.so').load_trilite_extension()
```

```
 56 /* Entry point for applications loading this module,
 57  * This will load the module */
 58 void load_trilite_extension(){
 59   sqlite3_auto_extension((void(*)(void))sqlite3_extension_init);
 60 }
```

The failing version is **1.0-kpegke2**, which has a symbol `sqlite3_extension_init` available in `libtrilite.so`.

**Comment by gartung** [comment]
On Oct 5, 2015, at 2:34 PM, davidlt <notifications@github.com<mailto:notifications@github.com>> wrote:

Could you explain in more details? How Darwin specific (or maybe it's not?) file is affecting Linux?

—
Reply to this email directly or view it on GitHubhttps://github.com/cms-sw/cmsdist/pull/1865#issuecomment-145644497.

OK. So it is not because of this line. My best guest is that libtrilite was built with sqlite 3.7 and the environment has sqlite 3.8. By rebuilding py2-dxr, libtrilite is rebuilt and linked against sqlite 3.8.

**Comment by davidlt** [comment]
Well, the Makefile is terrible. It basically links to everything on the system and completely ignores "CMSSW environment".  It's also using _pkg-config_.

**Comment by gartung** [comment]
On Oct 5, 2015, at 3:32 PM, davidlt <notifications@github.com<mailto:notifications@github.com>> wrote:

I hate this package, because it's a hacky mess. I think, I had to patch it again for Clang 3.7.0.

The master version of DXR now use ElasticSearch instead of libtrilite.so and sqlite3. It probably also has updates for Clang 3.7.0.
The problem is the requirement for an ElasticSearch server. I was playing around with the vagrant VM that runs an ElasticSearch server.

**Comment by davidlt** [comment]
We have ES server (or had) for CMSSW IB logs.

**Comment by gartung** [comment]
OK. I will submit an updated DXR spec that builds the master branch.

On Oct 5, 2015, at 3:47 PM, davidlt <notifications@github.com<mailto:notifications@github.com>> wrote:

We have ES server (or had) for CMSSW IB logs.

**Comment by gartung** [comment]
Because py2-dxr is becoming hard to maintain I propose we remove it from the cmssw-tool-conf.

**Comment by gartung** [comment]
@davidlt As long as a rebuild against sqlite 3.8 is forced the segfault goes away. I removed py2-dxr and its requirements from cmssw-tool-conf.spec but I would still like to keep the dxr build option.

**Comment by ghost** [comment]
@gartung As it is right now this PR cannot be merged. Please do a rebase so we can test it.

File: py2-dxr.spec [hunk 1]
```diff
@@ -1,8 +1,8 @@
 ### RPM external py2-dxr 1.0
 ## INITENV +PATH PYTHONPATH %i/$PYTHON_LIB_SITE_PACKAGES
 BuildRequires: llvm sqlite
-Requires: python zlib py2-setuptools py2-futures py2-jinja py2-markupsafe py2-ordereddict py2-parsimonious
-
+Requires: python llvm sqlite zlib py2-setuptools py2-futures py2-jinja py2-markupsafe py2-ordereddict py2-parsimonious
+%define isdarwin %(case %{cmsos} in (osx*) echo 1 ;; (*) echo 0 ;; esac)
```
Comment (davidlt) [inline py2-dxr.spec]: There is no need for this line.

**Comment by davidlt** [comment]
So why do we remove it? We only need to rewrite a small Makefile to look into correct places for us. Should be relatively easy.

**Comment by gartung** [comment]
Will resubmit with just the last change on a branch.

**Pull request closed without merging**